qiskit/pulse/schedule.py

```diff
@@ -811,14 +859,35 @@ def __call__(self, *args: List[Union[float, complex]],
         return self.bind_parameters(*args, **kwargs)
 
 
-def _insertion_index(intervals: List[Interval], new_interval: Interval, index: int = 0) -> int:
+def _find_interval(intervals: List[Interval], new_interval: Interval, index: int = 0) -> int:
```

Comment:
This points out that I wrote `insertion_index` to do too much -- it shouldn't be responsible for checking overlaps. The caller should get to decide what to do with the returned index.

It's clear that these two functions do the same thing, since their names contain the same information (just paraphrased). This should become one function again (maybe a better name is `_find_insertion_index` or `_find_interval_index`) and remain as it was, except that `if _overlaps(intervals[index], new_interval):` should be moved to `_add_timeslots`, which is where the error should have been raised in the first place.

Reply:
I think its fine to have this as two separate methods. They are definitely doing different things.

Reply:
The methods calling them are doing different things, but this function should be one function. This function isn't returning an interval, it's returning an index. If the interval exists in the input list, it's "insertion index" should be where it already is, so the result should be the same. 

I could be wrong though. Could you should me a few examples where
intervals: List[Tuple[int, int]]] and interval: Tuple[int, int]
and what the return value should be for `find_interval` vs `insertion_index`, given that the caller would handle overlapping and equality checking